### PR TITLE
Restore tests, add in-source testing

### PR DESCRIPTION
**Pull request by nedredmond**
Restores original tests by adding back some of the things cut from the KA module build, just for the test build.

These are not true unit tests-- they test the entire build codebase. Testing individually is difficult since we cannot import/export anything, so...

Also adds the capacity to do in-source testing with `vitest`, although we need to comment these out when we're not using them for MathQuill reasons. Long story short, MathQuill needs to exist in a pristine pre-modular state where imports/exports don't exist. If you introduce any of those to the ecosystem, you need to refactor the entire codebase.

In order to remove this code from the build without treeshaking, we have introduced `// rm_below_makefile` and `// rm_above_makefile`, a pair of delimiters that will strip out whatever is between them. Comments are already stripped out, but these have come in handy when restoring old window-manipulating behavior (see `MathQuill.noConflict` in `publicapi.ts`).

Issue: LC-1739

Testing:
That's what it's all about!

For the classic mocha tests:
1. Run `make test` to get all recent changes
2. Open `test/unit.html` in any browser

For the in-source tests:
1. Uncomment the tests you'd like to run, including their encapsulating conditional block. So far, the only unit tests are in `src/services/aria.ts`
3. Run `npm test`, an alias for `npx vitest`.

### CODE REVIEW

File: docs/Testing.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Testing
+
+## Integration tests
+
+To run the integration tests, run `make test` in the root.
+This will create a backwards-compatible build.
```
Comment (benchristel) [inline docs/Testing.md:6]: What does "backwards-compatible" mean here? Compatible with what?

**Review by benchristel — APPROVED** [review]
LGTM. Thanks for getting the tests working!

The "commented out unit tests" thing is pretty hacky, and I suspect the tests will seldom get run because of the manual work involved. I might take a crack at converting the vitests to https://github.com/benchristel/taste so they can run in the HTML page with the rest of the tests and we don't need to comment them out.

File: docs/Testing.md (hunk 1 above)
Comment (nedredmond) [reply docs/Testing.md:6]: Compatible with a not-JS-Module, I suppose?

Comment (benchristel) [reply docs/Testing.md:6]: Does `make test` overwrite build artifacts other than the HTML files?

Comment (benchristel) [reply docs/Testing.md:6]: If not, I think it might be clearer to say something like "This will generate HTML files in `test/`."

Comment (nedredmond) [reply docs/Testing.md:6]: Not the ones we consume. It overwrites `mathquill-basic.js`. The only one that is consumed through the package is the minified version of `mathquill.js`, and that one is unaffected.

**Pull request merged**